src/utils/generator.js

```diff
@@ -263,6 +263,10 @@ export class GeneratorEntry {
     return this.index < entry.index;
   }
 
+  getHash() {
```

Comment:
What about `getKey` or `getDedupeKey` instead of `getHash`?

Reply:
I'm not bothered on naming really, `getKey` or `getDedupeKey` is good with me too. I went with `getHash` just to get some code out there so others could get their teeth into the proposal.